integrations/langchain/src/unitycatalog_ai_langchain/toolkit.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging
+import os
+from typing import Any, Dict, List, Optional
+
+from langchain_core.pydantic_v1 import BaseModel, Field, root_validator
+from langchain_core.tools import StructuredTool, Tool
+from unitycatalog.ai.client import BaseFunctionClient, get_uc_function_client
+from unitycatalog.ai.utils import (
+    generate_function_input_params_schema,
+    validate_full_function_name,
+)
+
+_logger = logging.getLogger(__name__)
+
+UC_LIST_FUNCTIONS_MAX_RESULTS = 100
+
+
+def get_tool_name(func_name: str) -> str:
+    if len(func_name) > 64:
+        tool_name = func_name[-64:]
+        _logger.warning(
+            f"Function name {func_name} is too long, truncating to 64 characters {tool_name}."
+        )
+        return tool_name
+    return func_name
+
+
+def validate_or_set_default_client(client: Optional[BaseFunctionClient] = None):
+    client = client or get_uc_function_client()
+    if client is None:
+        raise ValueError(
+            "No client provided, either set the client when creating the "
+            "LangchainToolkit, or set the default client using "
+            "unitycatalog.ai.client.set_uc_function_client(client)."
+        )
+    return client
+
+
+class LangchainToolkit(BaseModel):
+    function_names: List[str] = Field(
+        default_factory=list,
+        description="The list of function names in the form of 'catalog.schema.function'",
+    )
+
+    tools_dict: Dict[str, Tool] = Field(default_factory=dict)
+
+    client: Optional[BaseFunctionClient] = Field(
+        default=None,
+        description="The client for managing functions, must be an instance of BaseFunctionClient",
+    )
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    @root_validator
+    def validate_toolkit(cls, values) -> Dict[str, Any]:
+        client = validate_or_set_default_client(values.get("client"))
+        values["client"] = client
+
+        function_names = values["function_names"]
+        tools_dict = values["tools_dict"]
+        for name in function_names:
+            if name not in tools_dict:
+                full_func_name = validate_full_function_name(name)
+                if full_func_name.function_name == "*":
+                    token = "None"
+                    while token is not None:
+                        functions = client.list_functions(
+                            catalog=full_func_name.catalog_name,
+                            schema=full_func_name.schema_name,
+                            max_results=os.environ.get(
+                                "UC_LIST_FUNCTIONS_MAX_RESULTS", UC_LIST_FUNCTIONS_MAX_RESULTS
+                            ),
+                            page_token=token if token != "None" else None,
+                        )
```

Comment:
what would happen if I had 1M function in my schema?

Reply:
Yea that's a problem, we can add a `max_functions` limit

Reply:
why do we need to fetch all of them? Can we just fetch the required functions?

Reply:
Emm what do you mean by `required functions`? If users use `catalog.schema.*` that means the user wants to include all functions in that schema; otherwise the user could specify the function name for a specific function

Reply:
For example, if `function_names` is `["foo", "bar"]`, I only need `foo` and `bar`, right?

Reply:
> If users use catalog.schema.* that means the user wants to include all functions in that schema

Got it :)